DDCore/src/segmentations/CylindricalGridPhiZ.cpp

```diff
@@ -0,0 +1,106 @@
+	registerParameter("grid_size_z",   "Cell size in Z",   _gridSizeZ,   1., SegmentationParameter::LengthUnit);
+	registerParameter("offset_phi", "Cell offset in phi", _offsetPhi,    0., SegmentationParameter::AngleUnit, true);
+	registerParameter("offset_z",   "Cell offset in Z",   _offsetZ,      0., SegmentationParameter::LengthUnit, true);
+	registerParameter("radius",     "Radius of Cylinder", _radius,       0., SegmentationParameter::LengthUnit);
+	registerIdentifier("identifier_phi", "Cell ID identifier for phi", _phiId, "phi");
+	registerIdentifier("identifier_z",   "Cell ID identifier for Z",   _zId,   "z");
+}
+
+/// Default constructor used by derived classes passing an existing decoder
+CylindricalGridPhiZ::CylindricalGridPhiZ(const BitFieldCoder* decode) :
+	CylindricalSegmentation(decode) {
+	// define type and description
+	_type = "CylindricalGridPhiZ";
+	_description = "Cylindrical segmentation in the local PhiZ-cylinder";
+
+	// register all necessary parameters
+	registerParameter("grid_size_phi", "Cell size in phi", _gridSizePhi, 1., SegmentationParameter::AngleUnit);
+	registerParameter("grid_size_z",   "Cell size in Z",   _gridSizeZ,   1., SegmentationParameter::LengthUnit);
+	registerParameter("offset_phi", "Cell offset in phi", _offsetPhi,    0., SegmentationParameter::AngleUnit, true);
+	registerParameter("offset_z",   "Cell offset in Z",   _offsetZ,      0., SegmentationParameter::LengthUnit, true);
+	registerParameter("radius",     "Radius of Cylinder", _radius,       0., SegmentationParameter::LengthUnit);
+	registerIdentifier("identifier_phi", "Cell ID identifier for phi", _phiId, "phi");
+	registerIdentifier("identifier_z",   "Cell ID identifier for Z",   _zId,   "z");
+}
+
+/// destructor
+CylindricalGridPhiZ::~CylindricalGridPhiZ() {
+
+}
+
+/// determine the position based on the cell ID
+Vector3D CylindricalGridPhiZ::position(const CellID& cID) const {
+	vector<double> localPosition(3);
+	Vector3D cellPosition;
+	double phi =
+	  binToPosition( _decoder->get(cID,_phiId ), _gridSizePhi, _offsetPhi);
+	cellPosition.Z =
+	  binToPosition( _decoder->get(cID,_zId ),   _gridSizeZ,   _offsetZ);
+	const double &R = _radius;
+	cellPosition.X = R*cos(phi); cellPosition.Y = R*sin(phi);
+
+	return cellPosition;
+}
+
+/// determine the cell ID based on the position
+  CellID CylindricalGridPhiZ::cellID(const Vector3D& localPosition, const Vector3D& /* globalPosition */, const VolumeID& vID) const {
+	double phi = atan2(localPosition.Y,localPosition.X);
+	double Z = localPosition.Z;
+	if ( phi < _offsetPhi) {
+	  phi += 2*M_PI;
+	}
+        CellID cID = vID ;
+        _decoder->set( cID,_phiId, positionToBin(phi, _gridSizePhi, _offsetPhi) );
+	_decoder->set( cID,_zId,   positionToBin(Z,   _gridSizeZ,   _offsetZ) );
+
+	return cID ;
+}
+
+std::vector<double> CylindricalGridPhiZ::cellDimensions(const CellID&) const {
+#if __cplusplus >= 201103L
+  return {_gridSizePhi, _gridSizeZ};
+#else
```

Comment:
We don't need the else anymore.

Reply:
I made a mistake here: line 94 should read "return {_radius*_gridSizePhi, _gridSizeZ};" so that the returned quantity be of the dimension of a length. I intend to commit that modification. Concerning the cancellation of the else, I would make the same remark as before: this kind of modification could instead be included in a dedicated commit that does the modification for all of the segmentation classes.

Reply:
grep -R "__cplusplus >= 201103L" returns 21 matches from different corner of the DD4hep distribution. I then do not feel like making any modification here.

Reply:
We haven't done clean up in other places of the code. Doesn't mean we need to introduce new pieces that we just need to clean up later. DD4hep doesn't compile with anything less than c++14, 

https://github.com/AIDASoft/DD4hep/blob/fe64884cac492f18574a8e2af37560667f7f3802/CMakeLists.txt#L59-L60

and we only probably test with c++17 or newer.

Reply:
I removed the #if,#else construct and committed.